### PR TITLE
CLI: Update Detector Configurations

### PR DESCRIPTION
**Pull request by VijayanB**
Update Detector Configurations

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by VijayanB** [comment]
Update Feature Workflow:
1) User will download detector configuration using download command. This is already merged.
2)User will make changes to the downloaded file
3) User will use update command to pass file name
4) if detector is updated after user downloaded file, cmd will fail with message please download latest version and merge
5)if detector is running, it will fail saying that detector is running.
6) User can either pass --force , cmd will automatically stop the detector and override the existing setting.
7) if user pass --restart, the detector will also start after successful update.

**Review by kaituo — APPROVED** [review]

File: cli/cmd/update.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	handler "esad/internal/handler/ad"
+	"fmt"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	commandUpdate = "update"
+	flagForce     = "force"
+	flagRestart   = "restart"
```
Comment (ylwu-amzn) [inline cli/cmd/update.go:25]: User can only "restart" a running detector? If current detector hasn't been started, is it possible user only set "restart" as true to start it after update ? 
From the description "Will start detector if update is successful", it's better to name the flag as "start".

File: cli/internal/controller/ad/ad.go [hunk 2]
```diff
@@ -520,3 +521,48 @@ func (c controller) GetDetectorsByName(ctx context.Context, pattern string, disp
 	}
 	return output, nil
 }
+
+//UpdateDetector updates detector based on DetectorID, if force is enabled, it overrides without checking whether
+// user downloaded latest version before updating it, if restart is true, detector will be started after update
```
Comment (ylwu-amzn) [inline cli/internal/controller/ad/ad.go:526]: Check detector is running or not if "force" is false?

File: cli/internal/controller/ad/ad.go [hunk 3]
```diff
@@ -520,3 +521,48 @@ func (c controller) GetDetectorsByName(ctx context.Context, pattern string, disp
...
+			fmt.Sprintf(
+				"esad will update detector: %s . Do you want to proceed? please type (y)es or (n)o and then press enter:",
+				input.ID,
+			),
+		),
+	)
+	if !proceed {
+		return nil
+	}
+	err := c.StopDetector(ctx, input.ID)
```
Comment (ylwu-amzn) [inline cli/internal/controller/ad/ad.go:552]: If "force" is false, no need to stop detector.

File: cli/internal/controller/ad/ad.go [hunk 4]
```diff
@@ -520,3 +521,48 @@ func (c controller) GetDetectorsByName(ctx context.Context, pattern string, disp
...
+	}
+	if !force {
+		latestDetector, err := c.GetDetector(ctx, input.ID)
+		if err != nil {
+			return err
+		}
+		if latestDetector.LastUpdatedAt > input.LastUpdatedAt {
+			return fmt.Errorf(
+				"new version for detector is available. Please fetch latest version and then merge your changes")
+		}
```
Comment (ylwu-amzn) [inline cli/internal/controller/ad/ad.go:539]: Check if detector is running? If detector is running and `force` is false, we should fail immediately.

File: cli/internal/controller/ad/ad.go (hunk 3 above)
Comment (VijayanB) [reply cli/internal/controller/ad/ad.go]: Ack

File: cli/internal/controller/ad/ad.go (hunk 4 above)
Comment (VijayanB) [reply cli/internal/controller/ad/ad.go:539]: Ack.
Screen Shot 2020-10-06 at 3 36 47 PM

File: cli/cmd/update.go (hunk 1 above)
Comment (VijayanB) [reply cli/cmd/update.go]: Ack

**Review by ylwu-amzn — APPROVED** [review]
LGTM. Thanks for the change!

**Pull request merged**